src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -99,7 +129,27 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 		if (!terminal || !commandDetection) {
 			return;
 		}
-		this._register(commandDetection.onCommandFinished(command => this._resolveQuickFixes(command)));
+		this._register(commandDetection.onCommandFinished(command => {
+			if (this._fixesAvailable) {
+				if (this._fixesShown) {
+					if (this._expectedCommands?.includes(command.command)) {
+						this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixExecutedManually);
+						this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixExecutedManually);
+					} else {
+						this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixWrong);
+						this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixWrong);
+					}
+					this._fixesShown = false;
+				} else {
+					this._logService.debug(QuickFixCommandResultTelemetryEvent.QuickFixIgnored);
+					this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>(QuickFixCommandResultTelemetryEvent.QuickFixIgnored);
+				}
+			} else {
+				this._expectedCommands = undefined;
+			}
+			this._resolveQuickFixes(command);
+			this._fixesAvailable = false;
```

Comment:
I think `_fixesAvailable` and `_expectedCommands` serve the same purpose and we could just use `_expectedCommands` instead (_fixesAvailable = true = !!_expectedCommands)

Reply:
This was necessary because there won't be expected commands in the `open` case